config/formats/typescriptExportDefinition.ts

```diff
@@ -117,7 +124,7 @@ const getTypeDefinition = (
   // get token type declaration from file
   const designTokenTypes: string[] = []
   for (const type of usedTypes) {
-    designTokenTypes.push(getTokenType(`${tokenTypesPath}/${type}.d.ts`))
+    designTokenTypes.push(getTokenType(`${tokenTypesPath.replace(new RegExp(/\/$/, 'g'), '')}/${type}.d.ts`))
```

Comment:
This is getting a bit long, and difficult to read. 

Can you break it up into multiple lines? And perhaps pull the regex out and either add annotations around it or write some tests for it so it's clear what it's doing?

Reply:
This ended up turning into quite a large refactor 🤣 

I refactored `toType` into a more generic `treeWalker` (now imported from another file) and removed the usedTypes as a side-effect. Instead I now run the treeWalker twice, once to build the object tree an once to get the used types.

I feel performance is not as important as clarity in this instance.

Reply:
I also now don't throw anymore as the tree walker ignores invalid items. It felt this was the correct approach, but let me know if you think it should throw (and why).